### PR TITLE
ADV7511 update

### PR DESCRIPTION
**Pull request by adrimbarean**
Update the ADV7511 no-OS project.

### CODE REVIEW

**Comment by adrimbarean** [comment]
V1:
 - fix Codacy errors.

File: projects/adv7511/src/main.c [hunk 1]
```diff
@@ -33,148 +35,370 @@
...
+		return -1;
+	return i2c_remove(i2c_mux);
 
-#ifdef XPAR_AXI_IIC_0_BASEADDR
-	HAL_PlatformInit(
-		XPAR_AXI_IIC_0_BASEADDR,	/* Perform any required platform init */
-		XPAR_SCUTIMER_DEVICE_ID,	/* including hardware reset to HDMI devices */
-		XPAR_SCUGIC_SINGLE_DEVICE_ID,
-		XPAR_SCUTIMER_INTR);
+//	ret = i2c_set_address(adv7511_i2c, mux_addr);
```
Comment (mchindri) [inline projects/adv7511/src/main.c:230]: should this be deleted?

**Comment by adrimbarean** [comment]
V2:
 - fix Doxygen warnings.

File: projects/adv7511/src/cf_hdmi.c [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /***************************************************************************//**
```
Comment (adrimbarean) [inline projects/adv7511/src/cf_hdmi.c:1]: Per @dbogdan 's comment use clk_axi_clkgen module here.

Comment (adrimbarean) [reply projects/adv7511/src/cf_hdmi.c:1]: Updated project to use clk_axi_clkgen module.

File: projects/adv7511/src/main.c (hunk 1 above)
Comment (adrimbarean) [reply projects/adv7511/src/main.c]: Deleted.

File: projects/adv7511/src/main.c [hunk 3]
```diff
@@ -33,148 +35,370 @@
...
+#define HDMI_CALL_INTERVAL_MS 10
 
 /******************************************************************************/
-/************************** Macros Definitions ********************************/
+/************************ Variables Definitions *******************************/
 /******************************************************************************/
-#define HDMI_CALL_INTERVAL_MS	10			/* Interval between two         */
-/* iterations of the main loop  */
-#define DBG_MSG                 xil_printf
+static uint8_t    major_rev;      /* Major Release Number */
```
[the remark this answers is not in the document]
Comment (adrimbarean) [reply projects/adv7511/src/main.c:76]: Added new line.

File: projects/adv7511/src/config.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+/**************************** Platform used ***********************************/
+/******************************************************************************/
+#define PLATFORM_AC701 0
+#define PLATFORM_KC705 1
+#define PLATFORM_VC707 2
+#define PLATFORM_ZC702 3
+#define PLATFORM_ZC706 4
+#define PLATFORM_ZED 5
+
+#define CURRENT_PLATFORM PLATFORM_ZC702
```
[the remark this answers is not in the document]
Comment (adrimbarean) [reply projects/adv7511/src/config.h]: Removed CURRENT_PLATFORM.

File: projects/adv7511/src/config.h [hunk 5]
```diff
@@ -0,0 +1,63 @@
+/***************************************************************************//**
+ *   @file   config.h
+ *   @brief  Config file of AD9361/API Driver.
+ *   @author DBogdan (dragos.bogdan@analog.com)
```
[the remark this answers is not in the document]
Comment (adrimbarean) [reply projects/adv7511/src/config.h]: Fixed.

File: projects/adv7511/src/config.h [hunk 6]
```diff
@@ -0,0 +1,63 @@
+/***************************************************************************//**
+ *   @file   config.h
```
[the remark this answers is not in the document]
Comment (adrimbarean) [reply projects/adv7511/src/config.h]: Renamed.

**Comment by adrimbarean** [comment]
V3 change log:
 - Use clk_axi_clkgen module in cf_hdmi module;
 - Delete commented code;
 - Rename config.h to app_config.h;
 - Remove CURRENT_PLATFORM define from app_config.h.

**Comment by adrimbarean** [comment]
V4 change log:
 - fix doxygen complaints.

File: projects/adv7511/src/edid.c [hunk 7]
```diff
@@ -0,0 +1,922 @@
...
+ *
+ * edid_parse_cea_block() helper function.
+ *
+ * @param [in] blk_ptr - Pointer to the data part in the EDID block.
+ * @param [in] len - Length of DBC area.
+ * @param [out] spa_ptr - Pointer to receive a pointer to the SPA.
+ *
+ * @return void
+ */
+static void edid_parse_data_blk_collection(uint8_t *blk_ptr, uint8_t len,
```
[the remark this answers is not in the document]
Comment (adrimbarean) [reply projects/adv7511/src/edid.c:489]: Done.

**Comment by adrimbarean** [comment]
V5 change log:
 - split EDID parse function into multiple smaller ones.

**Review by dbogdan — APPROVED** [review]

**Pull request merged**